crates/acir/src/circuit/gate.rs

```diff
@@ -31,6 +31,20 @@ pub enum Gate {
 }
 
 impl Gate {
+    pub fn name(&self) -> &str {
+        match self {
+            Gate::Arithmetic(_) => "arithmetic",
+            Gate::Range(_, _) => "range",
+            Gate::And(_) => "and",
+            Gate::Xor(_) => "xor",
+            Gate::Directive(Directive::Invert { x: _, result: _ }) => "invert",
+            Gate::Directive(Directive::Truncate { a: _, b: _, c: _, bit_size: _ }) => "truncate",
```

Comment:
To ignore all (or a subset of) fields in a struct pattern you can do `Directive::Invert { .. }`

Reply:
Cheers, I hadn't seen that before.